### DISCUSSION TITLE
Using the Read-Only toggle to toggle normal-mode and insert-mode

### DISCUSSION THREAD

**Discussion by honhimW** [JetBrains/ideavim#599] [Q&A]
Is it possible to use the Read-Only toggle to toggle normal-mode and insert-mode?
Or use the Read-Only toggle to enable/disable the plugin.

**Comment by ljchen1** [comment]
I think it is not possible.
But you could use the `imap <leader>f <ESC><action>(ToggleReadOnlyAttribute)` to map a shortcut to exit the insert mode and trigger the read-only toggle.
If you enable the "Do not show this dialog in the future" option below, when you use the `i` to go to the insert mode, it could make the file writeable directly which exits the Read-Only.

![image](https://user-images.githubusercontent.com/103916765/218934916-55c1c42c-2521-4a2b-8fae-b08291bd8a63.png)
